### PR TITLE
Search metadata and pagination support

### PR DESCRIPTION
**Pull request by adilansari**
#### Includes
- Client implementation for latest tigris search API
- Search pagination support
- Search hit metadata support

### CODE REVIEW

File: client/src/main/java/com/tigrisdata/db/client/TypeConverter.java [hunk 1]
```diff
@@ -176,6 +181,10 @@ public static Api.SearchRequest toSearchRequest(
     if (Objects.nonNull(req.getReadFields())) {
       builder.setFields(ByteString.copyFromUtf8(req.getReadFields().toJSON(objectMapper)));
     }
+    if (Objects.nonNull(options)) {
```
Comment (adilansari) [inline client/src/main/java/com/tigrisdata/db/client/TypeConverter.java:184]: setting pagination data in outgoing request.

File: client/src/main/java/com/tigrisdata/db/client/search/HitMeta.java [hunk 2]
```diff
@@ -14,19 +14,94 @@
...
+    this.updatedAt = updatedAt;
+  }
+
+  /**
+   * Gets the time at which document was inserted/replaced to a precision of milliseconds in {@code
+   * ZoneOffset.UTC}
+   *
+   * @return time in UTC or null
+   */
+  public OffsetDateTime getCreatedAt() {
```
Comment (adilansari) [inline client/src/main/java/com/tigrisdata/db/client/search/HitMeta.java:45]: DMLResponse exposes similar information as `protobuf.Timestamp`, should we instead present that as a Java native data type - `OffsetDateTime` or `Instant`?

File: client/src/main/java/com/tigrisdata/db/client/search/SearchMeta.java [hunk 3]
```diff
@@ -21,51 +21,51 @@
 public final class SearchMeta {
 
   private final long found;
-  private final long currentPage;
-  private final long totalPages;
-  private final int perPage;
+  private final int totalPages;
```
Comment (adilansari) [inline client/src/main/java/com/tigrisdata/db/client/search/SearchMeta.java:24]: Just a bit of restructuring as api changed from int64 to int32

File: client/src/main/java/com/tigrisdata/db/client/search/SearchRequestOptions.java [hunk 4]
```diff
@@ -16,12 +16,12 @@
...
 public final class SearchRequestOptions {
-  private static final long CURRENT_PAGE = 1;
-  private static final long DEFAULT_PER_PAGE = 10;
+  private static final int CURRENT_PAGE = 1;
+  private static final int DEFAULT_PER_PAGE = 20;
   private static final SearchRequestOptions DEFAULT_INSTANCE = newBuilder().build();
 
-  private final long page;
-  private final long perPage;
+  private final int page;
```
Comment (adilansari) [inline client/src/main/java/com/tigrisdata/db/client/search/SearchRequestOptions.java:23]: api changed from int64 to int32

File: client/src/main/java/com/tigrisdata/db/client/search/HitMeta.java (hunk 2 above)
Comment (JigarJoshi) [reply client/src/main/java/com/tigrisdata/db/client/search/HitMeta.java:45]: `Instant` sounds good 👍

File: client/src/main/java/com/tigrisdata/db/client/AbstractTigrisCollection.java [hunk 5]
```diff
@@ -100,7 +100,7 @@ protected Iterator<T> readInternal(
   protected Iterator<SearchResult<T>> searchInternal(
```
Comment (JigarJoshi) [inline client/src/main/java/com/tigrisdata/db/client/AbstractTigrisCollection.java:100]: this method is pulled upto Abstract class however it is not being used from anywhere other than StandardTigrisCollection. Can you localize it?

Comment (adilansari) [reply client/src/main/java/com/tigrisdata/db/client/AbstractTigrisCollection.java]: done

**Review by himank — APPROVED** [review]

**Pull request merged**